src/features/identityCheck/pages/identification/IdentityCheckUnavailable.tsx

```diff
@@ -3,19 +3,28 @@ import { useRoute } from '@react-navigation/native'
 import React from 'react'
 import styled from 'styled-components/native'
 
-import { navigateToHome, openUrl } from 'features/navigation/helpers'
+import { DMSModal } from 'features/identityCheck/components/DMSModal'
+import { navigateToHome } from 'features/navigation/helpers'
 import { UseRouteType } from 'features/navigation/RootNavigator'
-import { env } from 'libs/environment'
+import { analytics } from 'libs/analytics'
 import { ButtonPrimaryWhite } from 'ui/components/buttons/ButtonPrimaryWhite'
 import { ButtonTertiaryWhite } from 'ui/components/buttons/ButtonTertiaryWhite'
 import { GenericInfoPage } from 'ui/components/GenericInfoPage'
+import { useModal } from 'ui/components/modals/useModal'
 import { ExternalSite } from 'ui/svg/icons/ExternalSite'
 import { HappyFace } from 'ui/svg/icons/HappyFace'
 import { PlainArrowPrevious } from 'ui/svg/icons/PlainArrowPrevious'
 import { ColorsEnum, getSpacing, Spacer, Typo } from 'ui/theme'
 
 export function IdentityCheckUnavailable() {
   const { params } = useRoute<UseRouteType<'IdentityCheckUnavailable'>>()
+
+  const { visible, showModal, hideModal } = useModal(false)
+
+  const showDMSModal = () => {
```

Comment:
est-ce que dans ce cas, il faut utiliser [`useCallback`](https://fr.reactjs.org/docs/hooks-reference.html#usecallback) pour éviter les rendu inutiles ?

Reply:
En vérité je ne sais pas.

Reply:
tel que je le comprends : à chaque rendu, une nouvelle lambda sera assignée à `showDMSModal`, donc il fera le rendu du composant

tu peux le vérifier en activant Why Did You Render (décommentant l'`import './why-did-you-render'`)

https://github.com/pass-culture/pass-culture-app-native/blob/master/src/App.tsx

https://github.com/pass-culture/pass-culture-app-native/blob/master/src/App.web.tsx


tel que je l'imagine : `useCallback` devrait mémoïser la fonction, et renvoyer toujours la meme lambda tant que les dependances n'ont pas changées, ce qui permettrait d'éviter des rendu

Reply:
Mais je pense que si gain il y a, il est minime. [Article intéressant sur le sujet](https://kentcdodds.com/blog/usememo-and-usecallback)